test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -0,0 +1,186 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+from ptl.utils.pbs_crayutils import CrayUtils
+import os
+
+
+@tags('cray', 'smoke')
+class TestCraySmokeTest(TestFunctional):
+
+    """
+    Set of tests that qualifies as smoketest for Cray platform
+    """
+
+    def setUp(self):
+        if not self.du.get_platform().startswith('cray'):
+            self.skipTest("Test suite only meant to run on a Cray")
+        TestFunctional.setUp(self)
+
+    @tags('cray', 'smoke')
+    def test_cray_no_queued_job(self):
```

Comment:
> I have made this part of test_cray_login_job. 

Why only to test_cray_login_job, but not to the test_cray_compute_job test?  The tests can end up running in any order... I'm still unclear on what exactly you are trying to test for.  

> All jobs are deleted as part of setUp so it does not matter if any previous test that submitted jobs ran before this test.

But setUp will get run before test_cray_login_job too. In which case, what exactly are you testing for?  That setup did it's job before running 1 test, but not the other?

Maybe this verification move to setUp? Or be added to both tests? If the problem is a system setup problem, then we really only need that part to run once before the test_cray_login_job and test_cray_compute_job run. But if nodes were still queued was a PBS problem, then I suppose both tests should have the check.

Reply:
I have made this a separate test which tests for:
1. the number of nodes in State 'UP  B' (up and batch mode) should be the same number of compute nodes in pbsnodes.
2. if there are no queued jobs then there should not be any nodes in 'resv' and in 'use'.

I have updated pbs_crayutils.py to count nodes with State 'UP  B'.

Reply:
Hi @vccardenas, I believe we already have other PTL tests that verify the nodes in state up and batch mode have a corresponding vnode created for them.  See pbs_apbasil_inventory.py.  We don't need two tests doing the same thing.

As for part 2 of the test which checks for no queued jobs, and verifies that there are no nodes reserved and in use, this still seems like a system setup verification to me.  Maybe it should be added to the Cray specific setUp?  As a stand alone test that can run at anytime after any other test, I just don't see what value it adds.